### ISSUE TITLE
Valgrind error with format()

### ISSUE THREAD

**Issue by tomba** [HowardHinnant/date#215]
This piece of code
```
	sys_seconds t = now();
	std::stringstream ss;
	ss << format("%a", t);
```

causes Valgrind to print

```
==12968== Conditional jump or move depends on uninitialised value(s)
==12968==    at 0x60FD793: __strftime_internal (strftime_l.c:543)
==12968==    by 0x60FF835: strftime_l (strftime_l.c:459)
==12968==    by 0x5842298: std::__timepunct<char>::_M_put(char*, unsigned long, char const*, tm const*) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.22)
==12968==    by 0x5896CB1: std::time_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::do_put(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, tm const*, char, char) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.22)
==12968==    by 0x58952EC: std::time_put<char, std::ostreambuf_iterator<char, std::char_traits<char> > >::put(std::ostreambuf_iterator<char, std::char_traits<char> >, std::ios_base&, char, tm const*, char const*, char const*) const (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.22)
==12968==    by 0x1151EB: std::basic_ostream<char, std::char_traits<char> >& date::to_stream<char, std::char_traits<char>, std::chrono::duration<long, std::ratio<1l, 1l> > >(std::basic_ostream<char, std::char_traits<char> >&, char const*, date::fields<std::chrono::duration<long, std::ratio<1l, 1l> > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const*, std::chrono::duration<long, std::ratio<1l, 1l> > const*) (date.h:4590)
==12968==    by 0x114124: std::basic_ostream<char, std::char_traits<char> >& date::to_stream<char, std::char_traits<char>, std::chrono::duration<long, std::ratio<1l, 1l> > >(std::basic_ostream<char, std::char_traits<char> >&, char const*, std::chrono::time_point<std::chrono::_V2::system_clock, std::chrono::duration<long, std::ratio<1l, 1l> > > const&) (date.h:5621)
==12968==    by 0x1130B4: _ZN4date6formatIcNSt6chrono10time_pointINS1_3_V212system_clockENS1_8durationIlSt5ratioILl1ELl1EEEEEEEEDTcmcl9to_streamcl7declvalIRSt13basic_ostreamIT_St11char_traitsISB_EEEEfp_fp0_EtlNSt7__cxx1112basic_stringISB_SD_SaISB_EEEEEPKSB_RKT0_ (date.h:5645)
==12968==    by 0x10EAEA: main (main.cpp:238)
```

**Comment by HowardHinnant** [comment]
I've tried and failed to install valgrind on macOS.  Does this fix the error?
```
--- a/include/date/date.h
+++ b/include/date/date.h
@@ -4566,7 +4566,7 @@ to_stream(std::basic_ostream<CharT, Traits>& os, const CharT* fmt,
 {
     using namespace std;
     using namespace std::chrono;
-    tm tm;
+    tm tm{};
 #if !ONLY_C_LOCALE
     auto& facet = use_facet<time_put<CharT>>(os.getloc());
 #endif
```

**Comment by tomba** [comment]
Yes, it does.

**Comment by HowardHinnant** [comment]
Thanks!  Would you like to submit a P/R or would you prefer I do that?  I welcome contributors!

**Comment by tomba** [comment]
Sure, I'll do that.

**Comment by tomba** [comment]
Fixed by #216

**Comment by HowardHinnant** [comment]
Thanks!
